shared/rakefiles/xk.rake

```diff
@@ -0,0 +1,175 @@
+# what we want, i.e. don't create a new Key/key file when @gcp_creds_file
+# changes because of :configure_kubectl.
+@serviceaccount_key_file = ENV["TF_VAR_serviceaccount_key"]
+task :configure_serviceaccount => [@gcp_creds_file, :configure_current_project] do
+  # TODO: This command is duplicated from exekube's gcp-project-init (and
+  # hardcodes 'projectowner' instead of $SA_NAME which is only defined in
+  # gcp-project-init). If gcp-project-init becomes idempotent (GPII-2989,
+  # https://github.com/exekube/exekube/issues/92), or if this 'keys create'
+  # step moves somewhere else in exekube, call this command from that place
+  # instead.
+  sh "
+    [ -f $TF_VAR_serviceaccount_key ] || \
+    gcloud iam service-accounts keys create $TF_VAR_serviceaccount_key \
+      --iam-account projectowner@$TF_VAR_project_id.iam.gserviceaccount.com
+  "
+end
+
+@kubectl_creds_file = "/root/.kube/config"
+task :configure_kubectl => [@kubectl_creds_file]
+rule @kubectl_creds_file => [@gcp_creds_file] do
+  # This duplicates information in terraform code, 'k8s-cluster'
+  cluster_name = 'k8s-cluster'
+  # This duplicates information in terraform code, 'zone'. Could be a variable with some plumbing.
+  zone = 'us-central1-a'
+  sh "
+    existing_cluster=$(gcloud container clusters list --filter #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]})
+    if [ $? == 0 ] && [ \"$existing_cluster\" != \"\" ]; then \
+      gcloud container clusters get-credentials #{cluster_name} --zone #{zone} --project #{ENV["TF_VAR_project_id"]}
+    fi"
+end
+
+task :configure_current_project => [@gcp_creds_file] do
+  sh "gcloud config set project $TF_VAR_project_id"
+end
+
+# This task is being called from entrypoint.rake and runs inside exekube container.
+# It applies secret-mgmt, sets secrets, and then executes arbitrary command from args[:cmd].
+# You should not invoke this task directly!
+task :xk, [:cmd, :skip_secret_mgmt] => [@serviceaccount_key_file, @kubectl_creds_file] do |taskname, args|
+  @secrets = Secrets.collect_secrets()
+
+  sh "#{@exekube_cmd} up live/#{@env}/secret-mgmt" unless args[:skip_secret_mgmt]
+
+  Secrets.set_secrets(@secrets)
+
+  sh_filter "#{@exekube_cmd} #{args[:cmd]}" if args[:cmd]
+end
+
+task :refresh_common_infra, [:project_type] => [@gcp_creds_file] do | taskname, args|
+
+  next if args[:project_type] == "common"
+
+  # The common infrastructure might create the DNS resources, in order to avoid
+  # conflicts at the creation resources we need to import the already created
+  # DNS zone.
+
+  output = `#{@exekube_cmd} sh -c "\
+    terragrunt output -json dns_zones \
+    --terragrunt-working-dir /project/live/#{@env}/infra/network 2>/dev/null \
+    "`
+  begin
+    # What does !! mean?
+    # https://stackoverflow.com/questions/524658/what-does-mean-in-ruby
+    dns_zone_found = !!JSON.parse(output)
+  rescue JSON::ParserError
+    dns_zone_found = false
+  end
+
+  if not dns_zone_found then
```

Comment:
`unless` instead of `if not`, `then` is usually omitted in constructions like this in Ruby.